### ISSUE TITLE
Cannot setup - KeyError: 'timezone'

### ISSUE THREAD

**Issue by samhh** [petergridge/openweathermaphistory#7]
Hi there! I have this installed via HACS, and the following in `configuration.yaml`:

```yml
# [...]

sensor:
  - platform: openweathermaphistory
    name: rainfactor
    api_key: <API key validated via curl here>
```

On startup I see this in the logs:

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 673, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 775, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/openweathermaphistory/sensor.py", line 205, in async_added_to_hass
    await self._weatherhist.async_update()
  File "/config/custom_components/openweathermaphistory/weatherhistory.py", line 46, in async_update
    localtimezone = pytz.timezone(data["timezone"])
KeyError: 'timezone'
```

I have HASS 2022.8.5 installed in a container. The correct timezone is visible in the UI and is also present in `/config/.storage/core.config`.

Any ideas? Cheers!

**Comment by petergridge** [comment]
Hi, the data["timezone"] is getting the time zone attribute from the openweathermap api call result.

Use this is the call to use to validate if the timezone attribute is in your result:

https://api.openweathermap.org/data/2.5/onecall/timemachine?lat=66&lon=66&dt=1644966442&appid=yourapikey

1644966442 is the unix time you may need to change this, just use the current unix time from https://www.unixtimestamp.com/.
Put your api key in and copy the link into your browser.
You can replace the latitude and longitude data with your location if you want.

I have not seen this error before, this app has been stable for sometime, it uses version 2.5 of the API.

I will be offline for the next month but will see my emails, I will help where I can.

Cheers
Pete

**Comment by samhh** [comment]
That call doesn't work as I'm on the free plan, however the following does:

```console
$ curl -s 'https://api.openweathermap.org/data/2.5/weather/?lat=66&lon=66&appid=<API key>' | jq .timezone
18000
```

> I will be offline for the next month but will see my emails, I will help where I can.

Appreciate the help, no rush! :slightly_smiling_face:

**Comment by samhh** [comment]
Might this be an issue with new API keys?: https://github.com/home-assistant/core/issues/77795#issuecomment-1242233211

> Only the hourly mode will work with new generated free API keys

**Comment by petergridge** [comment]
You may be right, I will generate a new key when I get a chance and retest. It will be a pity if this is the case, there are alot of people who want to get to the history data.

**Comment by ColinRobbins** [comment]
I had this problem - I created a new key, and it did not work, with the same error.
Reverting to an old key, worked just fine.

**Comment by petergridge** [comment]
Thanks for the confirmation Colin

**Comment by thor0215** [comment]
Just to add to this.  I ran into the same issue and since I was new to this addon I did not have an old api key I could use.  Instead, I had to signup for the "One Call by Call" subscription plan.  This does require providing payment if you go over the daily limit of 1,000 calls a day.  I was curious if this would fix the timezone issue, which it did.  I also expected the chance that I go over the 1,000 calls a day to be low.  So far this fixes the timezone issue with new api keys, but the call to the 2.5 api does not seem to be recorded as a 3.0 api call so I do not have any daily usage yet.  I'm hoping this stays this way for us home users moving forward.
